dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/feedback/ImportReport.java

```diff
@@ -143,15 +172,47 @@ public Stats getStats()
     @JacksonXmlProperty( localName = "typeReport", namespace = DxfNamespaces.DXF_2_0 )
     public void setTypeReports( List<TypeReport> typeReports )
     {
+        typeReportMap.clear();
         if ( typeReports != null )
         {
             typeReports.forEach( tr -> typeReportMap.put( tr.getKlass(), tr ) );
         }
     }
 
-    public Map<Class<?>, TypeReport> getTypeReportMap()
+    public Set<Class<?>> getTypeReportKeys()
     {
-        return typeReportMap;
+        return unmodifiableSet( typeReportMap.keySet() );
+    }
+
+    @JsonIgnore
+    public int getTypeReportCount()
+    {
+        return typeReportMap.size();
+    }
+
+    public TypeReport getTypeReport( Class<?> klass )
+    {
+        return typeReportMap.get( klass );
+    }
+
+    public void forEachTypeReport( Consumer<TypeReport> reportConsumer )
+    {
+        typeReportMap.values().forEach( reportConsumer );
+    }
+
+    public ObjectReport getFirstObjectReport()
+    {
+        Iterator<TypeReport> iter = typeReportMap.values().iterator();
+        if ( !iter.hasNext() )
+        {
+            return null;
+        }
+        TypeReport report = iter.next();
+        if ( !report.hasObjectReports() )
+        {
+            return null;
+        }
+        return report.getFirstObjectReport();
```

Comment:
Note that this was how it was implemented before but it bags the question if the correct result would be to find the first `TypeReport` that has an `ObjectReport`. This simply assumes that the first `TypeReport` has one or the result is `null`.